roles/kubernetes-apps/metallb/tasks/main.yml

```diff
@@ -5,30 +5,21 @@
   when:
     - "kube_proxy_mode == 'ipvs' and not kube_proxy_strict_arp"
 
-- name: Kubernetes Apps | Check that the deprecated 'matallb_auto_assign' variable is not used anymore
-  fail:
-    msg: "'matallb_auto_assign' configuration variable is deprecated, please use 'metallb_auto_assign' instead"
-  when:
-    - matallb_auto_assign is defined
-
-- name: Kubernetes Apps | Lay Down MetalLB
-  become: true
-  template:
-    src: "metallb.yaml.j2"
-    dest: "{{ kube_config_dir }}/metallb.yaml"
-    mode: "0644"
-  register: metallb_rendering
-  when:
-    - inventory_hostname == groups['kube_control_plane'][0]
-
-- name: Kubernetes Apps | Install and configure MetalLB
```

Comment:
Don't you need to remove these older resources before install the helm chart when upgrading cluster? (or out of scope of this role?)

Reply:
If you delete the old resources directly, it will cause a service interruption, which I have seen someone else ask about earlier.
My idea is to add label and annotation to the existing MetalLB resources so that helm can take over.

wdyt?

Reply:
Understood.
No problem, if the charts can overwrite all of the old resources.